js/initcontext.go

```diff
@@ -56,20 +56,23 @@ type InitContext struct {
 
 	// Cache of loaded programs and files.
 	programs map[string]programWithSource
+
+	compatibilityMode compiler.CompatibilityMode
 }
 
 // NewInitContext creates a new initcontext with the provided arguments
 func NewInitContext(
-	rt *goja.Runtime, compiler *compiler.Compiler, ctxPtr *context.Context, filesystems map[string]afero.Fs, pwd *url.URL,
+	rt *goja.Runtime, c *compiler.Compiler, compatMode compiler.CompatibilityMode,
```

Comment:
I suppose `compiler` shadows the import and a linter complained ? I don't particularly like the name `c` though, but have no suggestions atm. 

Also .. It's strange they don't complain context.Context isn't the first argument, but than it's a pointer to context.Context which seems even fishier .... maybe I should create an issue and try to see if there are any *good* reasons for it being a pointer

Reply:
> I suppose `compiler` shadows the import and a linter complained ?

Right, but I think just my mental linter complained, since names shouldn't shadow imports like this. I know it made things awkward when I was first working on this feature.

I agree that `c` is not descriptive enough, but I think I'm only using it in limited scopes (i.e. not throughout a long function), so it should be fairly obvious what it refers to. I considered `comp`, but that has other connotations. I'm fine with leaving it as is.

Not sure about those issues with `Context`, haven't had any problems with it so far.

Reply:
Can we resolve this?